extract.go

```diff
@@ -0,0 +1,69 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+
+	"github.com/joho/godotenv"
+)
+
+type Url struct {
+	action, address, tag string
+}
+
+func newUrl(action, address, tag string) *Url {
+	return &Url{
+		action:  action,
+		address: address,
+		tag:     tag,
+	}
+}
+
+func getEtherscanKey() string {
+	// loads values from .env into the system
+	if err := godotenv.Load(); err != nil {
+		log.Fatal(err)
+	}
+	// gets ETHERSCAN_KEY
+	etherscanKey, exists := os.LookupEnv("ETHERSCAN_KEY")
+
+	if exists {
+		return etherscanKey
+	}
+	return ""
+}
+
+func buildUrl(action, address, tag string) string {
+	u := newUrl(action, address, tag)
+	return "https://api.etherscan.io/api?module=account&action=" + u.action + "&address=" + u.address + "&tag=" + u.tag + "&apikey=" + getEtherscanKey()
+}
+
+func call(url string) string {
+	response, err := http.Get(url)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	data, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return string(data)
+}
+
+// returns the Ether balance of a given address
+func getBalance(address, tag string) string {
+	return call(buildUrl("balance", address, tag))
+}
+
+func getBalances(address, tag string) string {
+	return call(buildUrl("balancemulti", address, tag))
+}
+
+func main() {
```

Comment:
can you add a comment to explain what this contract at `0xde0b295669a9fd93d5f28d9ec85e40f4cb697bae` does?

Reply:
ok got it. regarding conflicts, this is indeed annoying the fix all of them, but this is how you learn how to better work with git.

Reply:
What if I care only about the last commit, can I squash commits into one?

Reply:
Yes you can, but you just focus on rebasing first. We can squash commits as a next step.

Reply:
Anyway, commits will be squashed when you merge your PR to main.

Reply:
:rocket: